src/Middleware/Diagnostics/src/StatusCodePage/StatusCodePagesOptions.cs

```diff
@@ -26,21 +26,26 @@ public StatusCodePagesOptions()
 
             if (context.HttpContext.RequestServices.GetService<IProblemDetailsService>() is { } problemDetailsService)
             {
-                await problemDetailsService.WriteAsync(new ()
+                await problemDetailsService.WriteAsync(new()
                 {
                     HttpContext = context.HttpContext,
                     ProblemDetails = { Status = statusCode }
                 });
             }
 
```

Comment:
Checking the content-length and content-type is weird, just make the original code an else-if block.
```suggestion
            else if (!context.HttpContext.Response.HasStarted)
            {
                context.HttpContext.Response.ContentType = "text/plain";
                await context.HttpContext.Response.WriteAsync(body);
            }
```

Reply:
Oh, just saw your comment on the issue... I'll revisit this next week, but I still don't think we want to rely on checking content-length or content-type to determine the presence of a buffered body.

Reply:
Yeah! I don't like it as well but since we were using it already, I decided to suggest it. I tried a repro to see if I can find something reliable but as I mentioned, in the issue, I was not able to repro.